File: 2745-construct-the-longest-new-string/2745-construct-the-longest-new-string.rb
# @param {Integer} x
# @param {Integer} y
# @param {Integer} z
# @return {Integer}
def longest_string(x, y, z)
  ans = 0
  if x == y
    ans = 2 * [x, y].min
  else
    ans = 2 * [x, y].min + 1
  end

  ans += z

  return ans * 2
end

File: 2745-construct-the-longest-new-string/2745-construct-the-longest-new-string.js
/**
 * @param {number} x
 * @param {number} y
 * @param {number} z
 * @return {number}
 */
var longestString = function(x, y, z) {
    var ans = 0;
    if (x == y)
        ans = 2 * Math.min(x,y);
    else
        ans = 2 * Math.min(x,y) + 1;
            
    ans += z;

    return ans * 2;
};